### PR TITLE
Layer view hierarchy

### PR DESCRIPTION
**Pull request by donflopez**

### CODE REVIEW

**Comment by donflopez** [comment]
This fix #1593

File: src/geo/leaflet/leaflet-tiled-layer-view.js [hunk 1]
```diff
@@ -2,43 +2,59 @@ var _ = require('underscore');
...
+    self.trigger('loading');
+  });
+
+  this.leafletLayer.onAdd = function (map) {
+    L.TileLayer.prototype.onAdd.apply(this, arguments);
+    self._onAdd();
+  };
+};
+
+LeafletTiledLayerView.prototype = _.extend(
```
Comment (rochoa) [inline src/geo/leaflet/leaflet-tiled-layer-view.js]: A silly thing, as an alternative to `_.extend` using an empty object, we could use `_.defaults`. Next.

**Comment by rochoa** [comment]
A nice step forward, I would like to see some more changes similar to the way we inject the native map for the rest of the dependencies, instead of relying on `options` objects. But that's another story.

I would merge it and continue with similar approaches in next iterations 👍 .

**Comment by alonsogarciapablo** [comment]
Thanks for the CR @rochoa!

**Pull request merged**